// @octokit/action/dist-src/index.js
import { Octokit as Core } from "@octokit/core";
import { createActionAuth } from "@octokit/auth-action";
import { paginateRest } from "@octokit/plugin-paginate-rest";
import { legacyRestEndpointMethods } from "@octokit/plugin-rest-endpoint-methods";
import { VERSION } from "./version";
import { HttpsProxyAgent } from "https-proxy-agent";
const DEFAULTS = {
    authStrategy: createActionAuth,
    baseUrl: getApiBaseUrl(),
    userAgent: `octokit-action.js/${VERSION}`,
};
export const Octokit = Core.plugin(paginateRest, legacyRestEndpointMethods).defaults(function buildDefaults(options) {
    return {
        ...DEFAULTS,
        ...options,
        request: {
            agent: getHttpsProxyAgent(),
            ...options.request,
        },
    };
});
function getApiBaseUrl() {
    /* istanbul ignore next */
    return process.env["GITHUB_API_URL"] || "https://api.github.com";
}
function getHttpsProxyAgent() {
    const proxy = process.env["HTTPS_PROXY"] || process.env["https_proxy"];
    if (!proxy)
        return undefined;
    return new HttpsProxyAgent(proxy);
}
